adam/perception/high_level_semantics_situation_to_developmental_primitive_perception.py

```diff
@@ -156,12 +170,22 @@ def _perceive_property_assertions(self) -> None:
                     perceived_object = self._objects_to_perceptions[situation_object]
                     # Convert the property (which as an OntologyNode object) into PropertyPerception object
                     if BINARY in attributes_of_property:
-                        perceived_property = HasBinaryProperty(
+                        perceived_property: PropertyPerception = HasBinaryProperty(
                             perceived_object, property_
                         )
                     elif COLOR in attributes_of_property:
-                        # TODO: issue: generate perception of colors
-                        raise RuntimeError("Fix this issue COLOR property")
+                        # Sample an RGB value for the color property and generate perception for it
+                        if property_ in [RED, BLUE, GREEN, BLACK, WHITE]:
```

Comment:
```suggestion
                        if property_ in COLORS_TO_RGBS:
```

That way you don't need to update this code when new colors are added.

Reply:
Wow I can't believe I missed this. Thanks, just fixed it.